backend/src/main/java/com/example/dontwastetomuch/game/GameController.java

```diff
@@ -37,24 +38,36 @@ public List<Game> getAllGames(){
     }
 
     @GetMapping("/{gameId}")
-    public Game getOneGame(@PathVariable String gameId){
-        return gameService.getOneGame(gameId);
+    public UserGameDTO getOneGame(@PathVariable String gameId, Principal principal){
+        MyUser myUser = myUserRepo.findById(principal.getName()).orElseThrow();
+        UserGameDTO userGameDTO = new UserGameDTO();
+        userGameDTO.setUsername(myUser.getUsername());
+        userGameDTO.setGameName(gameService.getOneGame(gameId).getGameName());
+        userGameDTO.setPlaytime(myUser.getGameData().stream().filter(gameData ->  gameId.equals(gameData.getGameId())).findAny().orElseThrow().getPlaytime());
+        userGameDTO.setSpentMoney(myUser.getGameData().stream().filter(gameData -> gameId.equals(gameData.getGameId())).findAny().orElseThrow().getMoney());
+        userGameDTO.setGameId(gameId);
+        userGameDTO.setApproved(gameService.getOneGame(gameId).isApproved());
```

Comment:
Außerdem bastelst du hier das Ganze Objekt im Controller zusammen. Wenn du das in den Service packst, wird das mit dem Testen auch bissl einfacher